### ISSUE TITLE
Add instrumentation for the "go.opentelemetry.io/otel/trace".nonRecordingSpan

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go-instrumentation#974]
A common instrumentation pattern for tracing is to return the the `TracerProvider` from the span held in a context:

```go
import (
	"net/http"

	"go.opentelemetry.io/otel/trace"
)

func handler(w http.ResponseWriter, r *http.Request) {
	ctx := r.Context()
	tracer := trace.SpanFromContext(ctx).TracerProvider().Tracer("tracer")
	ctx, span := tracer.Start(ctx, "span")
	defer span.End()
	/* ... */
}
```

When using the instrumentation of the global trace API, this span will not be recorded.

**Describe the solution you'd like**

The span created in this above code should be captured by auto-instrumentation. Meaning the [`nonRecordingSpan`](https://github.com/open-telemetry/opentelemetry-go/blob/1b5834f96c6eaaac5ffc21250edc296dea2525df/trace/nonrecording.go#L9) should be instrumented.

**Comment by MrAlias** [comment]
This was only partially resolved by #1405. If a global span is already active in the context this will work, as demonstrated in #1405. However, if there is no active span a no-op span is still created and is not instrumented.

We need to provide the same enablement logic as the global API for the `trace.SpanFromContext` function.

**Comment by MrAlias** [comment]
My initial idea was to update the `noopSpan.TracerProvider` method:

https://github.com/open-telemetry/opentelemetry-go/blob/784638358b3d39412703a13fe0bd6aa0908ffe45/trace/noop.go#L84-L85

Similar to https://github.com/open-telemetry/opentelemetry-go/pull/5920 we could call a private method with a `*bool`. For example:

```diff
--- a/trace/noop.go
+++ b/trace/noop.go
@@ -6,6 +6,7 @@ package trace // import "go.opentelemetry.io/otel/trace"
 import (
        "context"

+       "go.opentelemetry.io/auto/sdk"
        "go.opentelemetry.io/otel/attribute"
        "go.opentelemetry.io/otel/codes"
        "go.opentelemetry.io/otel/trace/embedded"
@@ -82,4 +83,22 @@ func (noopSpan) AddLink(Link) {}
 func (noopSpan) SetName(string) {}

 // TracerProvider returns a no-op TracerProvider.
-func (noopSpan) TracerProvider() TracerProvider { return noopTracerProvider{} }
+func (s noopSpan) TracerProvider() TracerProvider {
+       return s.tracerProvider(autoInstEnabled)
+}
+
+// autoInstEnabled defines if the auto-instrumentation SDK is enabled.
+//
+// The auto-instrumentation is expected to overwrite this value to true when it
+// attaches to the process.
+var autoInstEnabled = new(bool)
+
+// tracerProvider return a noopTracerProvider if autoEnabled is false,
+// otherwise it will return a TracerProvider from the sdk package used in
+// auto-instrumentation.
+func (noopSpan) tracerProvider(autoEnabled *bool) TracerProvider {
+       if *autoEnabled {
+               return sdk.TracerProvider()
+       }
+       return noopTracerProvider{}
+}
```

This will not work because it will create an import cycle.

**Comment by MrAlias** [comment]
copying the `auto/sdk` to some `go.opentelemetry.io/otel/trace/internal` package would also have the same import cycle issue.

**Comment by MrAlias** [comment]
https://github.com/open-telemetry/opentelemetry-go/pull/6203 has merged. We can no instrument that copy of the SDK to support this.

**Comment by MrAlias** [comment]
Blocked by either a release of opentelemetry-go or the resolution of https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1753. Currently, when trying to test the added probe the comparison of `1.34.1-0.20250205150531-85fab8be9320 > 1.34.0` is returning `false` which means the probe package constraint will not work.

**Comment by RonFed** [comment]
@MrAlias Is the plan to remove the `sdk` package now that we moved it to the otel-go repo?
If so, should #1961 be done in otel-go?

**Comment by MrAlias** [comment]
No, the plan is to always maintain both in both locations. Any changes to the package here need to be PR-ed to `otel`.

**Comment by RonFed** [comment]
@MrAlias sorry if we already discussed this and I forgot,
Why do we need to keep the copy here in addition to the one in `otel-go`?

**Comment by MrAlias** [comment]
The package exported here is used in the global package (https://github.com/open-telemetry/opentelemetry-go/blob/838643a3a52415ca36f7e831e2edcc8c9d489d46/internal/global/trace.go#L152-L176). The one in the trace API is used there to avoid package import cycles. The trace API auto SDK is not exported (by design) and cannot be used anywhere else.

**Comment by MrAlias** [comment]
After https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/2001 is merged, we need to update dependency on `go.opentelemety.io/otel` to be a released (not commit hash) version. This will also require a release from that package.

**Comment by MrAlias** [comment]
Closing this as this as the instrumentation is working. Only thing left for clean up is waiting on upstream to release. Renovate will handle the upgrade when that happens.
